cobc/pplex.l

```diff
@@ -2592,8 +2599,38 @@ pp_text_list_add (struct cb_text_list *list, const char *text,
 	return list;
 }
 
+static struct cb_token_list *
+pp_token_list_add (struct cb_token_list *list,
+		      const char *text,
+		      const char *token)
+{
+	struct cb_token_list	*p;
+	void			*tp;
+	int text_size = strlen (text);
+
+	p = cobc_plex_malloc (sizeof (struct cb_token_list));
+	if (token == NULL) {
+		tp = cobc_plex_malloc (text_size + 1);
+		p->token = NULL;
+	} else {
+		int token_size = strlen (token);
+		tp = cobc_plex_malloc( text_size + token_size + 2);
```

Comment:
My understanding is that everything allocated during the preprocessing phase is freed in a single pass once the `.i` file has been generated, and nothing survives, so both data have the same lifetime expectancy.

Since there was a comment that a lot of time was spent in this part of the compiler, I tried to optimize the code with this single allocation and one removing one copy, though this code is probably almost very rarely executed...

Reply:
I understand that idea, but yes, that should be used rarely in any case so going for "more clear" is better.
And yes, that memory is freed all at once.

Reply:
ok, I used `cobc_plex_strdup` in the latest version.